tests/conftest.py

```diff
@@ -125,7 +125,25 @@ def run_and_parse(testdir, exit_code_exp=0, runpytest_args=None):
     return junit_xml_doc
 
 
-def run_and_parse_with_config(testdir, config, exit_code_exp=0, runpytest_args=None):
+def build_property_list(config_dict):
+    """ Build a list of properties and values that will end up in junit xml. This should only
+        be used for testing purposes to set expectations.
+
+    Args:
+        config_dict: a python object representation of pytest_zigzag config json
+
+    Returns:
+        props: a flattened dictionary of properties and values
+    """
+    props = {}
+    for root_key in config_dict:
+        if not config_dict[root_key] is None:
+            for key in config_dict[root_key]:
+                props[key] = config_dict[root_key][key]
+    return props
+
+
+def run_and_parse_with_config(testdir, config, exit_code_exp=0, runpytest_args=None, ini_config=None):
```

Comment:
Should this function be altered so that it can test scenarios with environment variables?

Reply:
That ability is already there due to the order of precedence when the config file is parsed, but that brings up a good point.  There should be a test for that.

Reply:
Test added.